webapp/packages/plugin-data-viewer/src/DatabaseDataModel/Actions/ResultSet/ResultSetSortAction.ts

```
@@ -0,0 +1,112 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2021 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { ResultDataFormat } from '@cloudbeaver/core-sdk';
+
+import { DatabaseDataAction } from '../../DatabaseDataAction';
+import type { IDatabaseDataOptions } from '../../IDatabaseDataOptions';
+import type { IDatabaseResultSet } from '../../IDatabaseResultSet';
+import { databaseDataAction } from '../DatabaseDataActionDecorator';
+import type { IDatabaseDataSortAction } from '../IDatabaseDataSortAction';
+import type { IResultSetElementKey } from './IResultSetElementKey';
+
+export enum ESortMode {
+  'asc' = 'asc',
+  'desc' = 'desc'
+}
+
+export type SortMode = ESortMode | null;
+
+@databaseDataAction()
+export class ResultSetSortAction extends DatabaseDataAction<IDatabaseDataOptions, IDatabaseResultSet>
+  implements IDatabaseDataSortAction<IResultSetElementKey, IDatabaseResultSet> {
+  static dataFormat = ResultDataFormat.Resultset;
+
+  private clear() {
+    if (!this.source.options) {
+      return;
+    }
+
+    this.source.options.constraints = [];
+  }
+
+  private deleteConstraint(columnName: string) {
+    if (!this.source.options) {
+      return;
+    }
+
+    this.source.options.constraints = this.source.options.constraints
+      .filter(constraint => constraint.attribute !== columnName);
+  }
+
+  private reorderConstraints() {
+    if (!this.source.options) {
+      return;
+    }
+
+    this.source.options.constraints = this.source.options.constraints
+      .map((constaint, idx) => ({ ...constaint, orderPosition: idx }));
+  }
+
+  setSortMode(columnName: string, sortMode: SortMode, multiple: boolean): void {
+    if (!this.source.options) {
+      return;
+    }
```

better to throw an exception, because we can't execute function correctly